src/iop/clipping.c

```diff
@@ -1615,9 +1616,101 @@ static void aspect_presets_changed(GtkWidget *combo, dt_iop_module_t *self)
           dt_bauhaus_combobox_set(combo, 0);
           return;
         }
-        d = dd;
-        n = nn;
+        d = MAX(dd, nn);
+        n = MIN(dd, nn);
       }
+      else
+      {
+        // find the closest fraction from the input ratio
+        float rr = atof(text);
+        int dd = ceilf(rr);
+        int nn = floor(rr);
+
+        // some sanity check
+        if(dd == 0)
+        {
+          dt_control_log(_("invalid ratio format. it should be greater than one"));
```

Comment:
greater than zero? or just "positive"

Reply:
Also it doesn't look like you're checking for negative values either.

Reply:
ooops, good catch.

Reply:
the only blocking case is if ==0, negative values are not an issue.

Reply:
edit : negative values make it joyfully crash, but that wasn't handled in the previous code either.